src/functions.sh

```diff
@@ -424,7 +424,7 @@ do
     # reset sudo
     sudo -k
     # check sudo
-    echo -e "$my_password\n" | sudo -Sv > /dev/null 2>&1
+    printf '%s\n' "$my_password" | sudo -Sv > /dev/null 2>&1
```

Comment:
This handles the edge case of the password containing an escape sequence such as `\n`